zerver/actions/streams.py

```diff
@@ -254,7 +254,14 @@ def do_unarchive_stream(
     )
 
     # Update caches
-    cache_set(display_recipient_cache_key(stream.recipient_id), new_name)
+    cache_set(
+        display_recipient_cache_key(stream.recipient_id),
+        {
+            "tag": "stream-recipient-info",
+            "stream_name": stream.name,
+            "archived": stream.deactivated,
+        },
+    )
```

Comment:
I still don't understand this change -- why do we need to mark stream messages as "archived" at all? I don't think it'll be possible to send a message to an archived stream, and clients should be able to look up via the `stream` data structures  whether a given stream that they are processing a message from is archived.

I suspect we just don't need this commit, but maybe I'm missing something.

Reply:
Ah I see. I misunderstood the situation and now  have made the necessary changes. 
The `archived` property has been removed from messages, and the client now checks the stream's archived status to display the "(archived)" postfix for deactivated streams.